qa/integration-tests/src/test/java/io/camunda/it/migration/util/CamundaMigrator.java

```diff
@@ -0,0 +1,288 @@
+            : opensearchConfiguration87(internalUrl);
+    if (envOverrides != null) {
+      env.putAll(envOverrides);
+    }
+    env.forEach(camundaContainer::withEnv);
+
+    camundaContainer.withCreateContainerCmdModifier(
+        cmd ->
+            cmd.withUser("1001:0")
+                .getHostConfig()
+                /* On camunda/camunda image the default Zeebe dataDir is /usr/local/camunda/data */
+                .withBinds(volume.asBind("/usr/local/camunda/data")));
+
+    camundaContainer.start();
+    url =
+        URL.formatted(
+            camundaContainer.getHost(), camundaContainer.getMappedPort(TestZeebePort.REST.port()));
+    camundaClient =
+        CamundaClient.newClientBuilder()
+            .grpcAddress(
+                URI.create(
+                    RPC_URL.formatted(
+                        camundaContainer.getHost(),
+                        camundaContainer.getMappedPort(TestZeebePort.GATEWAY.port()))))
+            .restAddress(URI.create(url))
+            .usePlaintext()
+            .build();
+
+    try {
+      login();
+    } catch (final IOException | InterruptedException e) {
+      throw new RuntimeException(e);
+    }
+
+    return this;
+  }
+
+  public CamundaMigrator update(final Map<String, String> envOverrides) {
+    /* Trigger snapshot of Zeebe's data to force flush ExporterMetadata */
+    PartitionsActuator.of(camundaContainer).takeSnapshot();
+
+    camundaContainer.close();
+    extractVolume();
+    broker =
+        new TestSimpleCamundaApplication()
+            .withBasicAuth()
+            .withAuthenticatedAccess()
+            .withAdditionalProfile(Profile.PROCESS_MIGRATION)
+            .withBrokerConfig(
+                cfg -> {
+                  cfg.getExperimental().setVersionCheckRestrictionEnabled(false);
+                  cfg.getGateway().setEnable(true);
+                })
+            .withWorkingDirectory(zeebeDataPath.resolve("usr/local/zeebe"));
+
+    final var multiDbConfigurator = new MultiDbConfigurator(broker);
+    if (databaseType.equals(DatabaseType.ES) || databaseType.equals(DatabaseType.LOCAL)) {
+      multiDbConfigurator.configureElasticsearchSupportIncludingOldExporter(
+          databaseUrl, indexPrefix);
+    } else {
+      multiDbConfigurator.configureOpenSearchSupportIncludingOldExporter(
+          databaseUrl, indexPrefix, OS_USER, OS_PASSWORD);
+    }
+    final Map<String, String> env = new HashMap<>();
+    env.put("camunda.migration.process.importerFinishedTimeout", "PT2S");
+    // Reduce importer intervals to speed up tests
+    env.put("camunda.operate.importer.importPositionUpdateInterval", "200");
+    env.put("camunda.operate.importer.readerBackoff", "200");
+    env.put("camunda.tasklist.importer.importPositionUpdateInterval", "200");
+    env.put("camunda.tasklist.importer.readerBackoff", "200");
+
+    env.putAll(envOverrides);
+    env.forEach(broker::withProperty);
+
+    broker.start();
+    broker.awaitCompleteTopology();
```

Comment:
🔧 Can we rename it to camunda or something, as it was a bit confusing (because I thought while reading we only spin up the broker)